Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+  in a multi-cluster setup.
+  However, there's folks that uses a Kubernetes Job to adds Silences,
+  In both situation, a CI/CD pipeline is needed in order to automate the execution, this includes managing the secrets and/or deprecation of code
+  In the Kubernetes Job way, teams need to build and deploy a container and monitor its status, which is an added efforts
+  CI/CD pipeline is not the best option in multi-cluster setups, teams need to manage k8s access, secrets, and configs.
+
+![CI/CD K8s Job Approach](../img/CICD-k8s-job.png "CI/CD K8s Job Approach")
+
+* Users that use a standalone operator that implements a Silence CRD are in a better situation,
+  for example users have reported that they are using GitOps to fully manage the life cycle of Silences Custom resources,
+  which in fact brings benefits to the whole team because having a history and revisions in Git as well as constancy to manage the silences in a multi-tenant environment
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that would require us to fully manage this component as of any managed app (monitoring, logging, alerting etc..).
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
+
+By Adding support for `Silence` CRD in the prometheus-operator, this will make users more flexible in terms of choosing the tool
+to deploy their silences and will free others from managing a standalone component within the stack.
+
+## Pitfalls of the current solution
+
+Using Alertmanager API Directly comes with drawbacks:
+
+* Teams have to build an automation to add silences in a centralized manner
+* There is no input validation, which can lead to an invalid silence configuration
+* Teams need to manage expiration on their own
+* If a silence was deleted by accident, it will be permanent.
+
+# Goals
+
+* Provide a way for users to manage Alertmanager silences with Custom Resources (CR).
+* Enable management of Alertmanager silences via centralised repo with e.g. ArgoCD, Flux...
+* Multi tenancy: Silences CR can be using kubernetes rbac.
+* Moving the Giant Swarm silence operator to prometheus operator would provide better community support
+
+## Audience
+
+* Users who serve Prometheus as a service and want to have an interface in defining silences exposed to developers.
+* Users who want to manage silences the same way as for services running within the Kubernetes cluster.
+* Users who want a supported Kubernetes way of silences outside the Kubernetes cluster
+
+# Non-Goals
+
+* This proposal do not cover automatic cleanup of expired Silence CRs
```

Comment:
expired silences are eventually removed by Alertmanager

Reply:
yes, But the CRs will remain in the cluster.

Reply:
Well removing them once they expire should be easy but if you use flux, you would need to have an automation in place to remove expired silences. Maybe CR removal can be disabled?

Reply:
In the first release, we will not cover automatic cleanup, maybe we implement it, if requested